File: 3. Time Complexity/PermMissingElem/PermMissingElem.cpp
#include <iostream>
#include <vector>
#include <algorithm>
#include<numeric>


using namespace std;

int solution(vector<int> &A)
{
    return (((A.size()+1)*(A.size()+2)/2) - (accumulate(A.begin(),A.end(),0)));
}

int main()
{
    vector<int> A = {2,3,1,5};
    solution(A);
}

File: 3. Time Complexity/PermMissingElem/PermMissingElem.java
class PermMissingElem 
{
    public int solution(int[] A)
    {
        long sum = 0;
        for(int i: A)
        {
            sum += i;
        }
        return (int)((((long)A.length+1)*((long)A.length+2)/2) - sum);
    }
}

